app/src/main/java/ch/epfl/sweng/zuluzulu/Structure/Post.java

```diff
@@ -19,19 +23,39 @@
     private String color;
     private int nbUps;
     private int nbResponses;
+    private List<String> upScipers;
+    private List<String> downScipers;
+    String id;
+
+    private boolean upByUser;
+    private boolean downByUser;
 
     private boolean anonymous;
+    private Channel channel;
+    private AuthenticatedUser userReading;
 
-    public Post(FirebaseMapDecorator data) {
+    public Post(FirebaseMapDecorator data, AuthenticatedUser userReading, Channel channel) {
```

Comment:
I know they have done it in the Association and Event class but still...

A structure class need to represent some thing. Like a Post. You don't want database logic here. You can create an external class to do the database request and to return a new Post (as a Builder).

Reply:
@Dahnepfl 
Users and channels are also structure classes, why should I not use them as parameters ? What do you mean by database logic here ?

Reply:
A Post is for instance represented by : a message, an user, a channel, a date etc...
But not by a FirebaseDecorator.

If I want to create a Post in a test for instance, or from a bot for an event. I shouldn't have to create a DatabaseDecorator to instanciate a Post.
I can still create a second constructor `Post(message, date, user, channel, ...) `. 
Mais à mon avis c'est mieux si on a un Proxy qui crée un `Post(message, date, user, channel, ...) `, comme ça tout ce qui a un rapport avec la base de donnée est dans le proxy et pas répartie un peu partout dans le code.
Du genre si tu veux récupérer la liste des posts, tu pourrais avoir :
```
proxy = new Proxy(databse)
List<Posts> = proxy.getPosts();
```
et dans proxy.getPosts() :
```
[...]
message = data.getString("message");
time = data.getDate("time");	       
list.add(new Post(time, message, user, channel, ...)
[...]
```

Reply:
@Dahnepfl 
Okay I see what you think that here. In reality I just need the user sciper and the channel id, I changed it.

Reply:
Ouais haha, je trouve ça mieux de créer un Post avec chaque paramètre en argument.
Désolé, j'arrête l'anglais xd

Du genre ici 

```
collectionReference
                .orderBy("time", Query.Direction.DESCENDING)
                .get()
                .addOnCompleteListener(new OnCompleteListener<QuerySnapshot>() {
                    @Override
                    public void onComplete(@NonNull Task<QuerySnapshot> task) {
                        if (task.isSuccessful()) {
                            posts.clear();
                            for (DocumentSnapshot document : task.getResult()) {
                                Log.d(TAG, document.getId() + " => " + document.getData());
                                FirebaseMapDecorator fmap = new FirebaseMapDecorator(document);
                                if (fmap.hasFields(Post.FIELDS)) {
                                    Post post = new Post(fmap, user, channel);
                                    posts.add(post);
                                }
                            }
                            adapter.notifyDataSetChanged();
                        } else {
                            Log.w(TAG, "Error getting documents.", task.getException());
                        }
                    }
                });
```

Tu pourrais deja changer ça par un 
`List<Post> = proxy.getPosts()`
après tu peux boucler dessus pour les afficher par exemple

Et ça

```
 for (DocumentSnapshot document : task.getResult()) {
                                Log.d(TAG, document.getId() + " => " + document.getData());
                                FirebaseMapDecorator fmap = new FirebaseMapDecorator(document);
                                if (fmap.hasFields(Post.FIELDS)) {
                                    Post post = new Post(fmap, user, channel);
                                    posts.add(post);
                                }
                            }
```

Tu pourrais faire (dans proxy.getPosts())


```
 for (DocumentSnapshot document : task.getResult()) {
                                Log.d(TAG, document.getId() + " => " + document.getData());
                                FirebaseMapDecorator fmap = new FirebaseMapDecorator(document);

                                message = data.getString("message");
                                time = data.getDate("time");	
                                Post post = new Post(time, message, user, channel);
                                 posts.add(post);
                                
                            }
```

Après c'est mon avis c'est pas très important, et je dis peut être de la merde haha

Reply:
Haha mais le FirebaseDecorator a justement été écrit dans le but de simplifier la création d'association, d'évent, etc... 
Dans le code tu lui donnes un snapshot, et dans les tests simplement une map.
En fait je vois pas du tout comment tu veux concrètement changer ça et pourquoi c'est un problème, notamment pour le proxy.

Reply:
Dans le cas présent je sais pas mais moi quand je crée une liste d'event dans le memento par exemple.

Je récupère les données en Json du site memento.epfl.ch
donc 
`title = jsonobject.getString("title")` par exemple

Et ce firebase decorator **m'oblige** à créer une Map pour créer un Event...

Alors que je pourrai simplement faire `new Event(title, date, etc...)`

Du coup je vais devoir créer un deuxième constructeur Event(title, date, etc...). 

Après tu peux laisser comme ça c'est égal

Reply:
Ah tu veux pouvoir donner simplement les arguments les uns après les autres.
Bon j'ai qu'à faire un constructeur de plus quoi ahah.
Désolé de t'avoir demandé de longues explications mais je voyais pas clairement ce que tu voulais haha